### ISSUE TITLE
Segfault when initializing in a brand new Ruby 3.0.0 / Rails 6.1.1 project

### ISSUE THREAD

**Issue by shepmaster** [sorbet/sorbet#3984]
#### Input

```
% rails new repro

% cd repro

% cat <<EOF >> Gemfile
gem 'sorbet', '~> 0.5', group: :development
gem 'sorbet-runtime', '~> 0.5'
EOF

% bundle

% bundle exec srb typecheck -e 'puts "Hello, world!"'
No errors! Great job.

% DISABLE_BOOTSNAP=1 srb init
```

#### Observed output

```
Generating: sorbet/config
Reusing existing config file: sorbet/config
Generating: sorbet/rbi/sorbet-typed/
Generating: sorbet/rbi/gems/
/Users/shep/.rvm/gems/ruby-3.0.0/gems/activesupport-6.1.1/lib/active_support/inflector/methods.rb:274: [BUG] Segmentation fault at 0x00007f82a0000028
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin20]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   for more details.
Don't forget to include the above Crash Report log file in bug reports.

-- Control frame information -----------------------------------------------
c:0036 p:---- s:0170 e:000169 CFUNC  :const_get
c:0035 p:0029 s:0165 E:000130 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/activesupport-6.1.1/lib/active_support/inflector/methods.rb:274
c:0034 p:0016 s:0159 E:001d60 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/activesupport-6.1.1/lib/active_support/core_ext/string/inflections.rb:74
c:0033 p:0103 s:0155 E:0014d0 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:176 [FINISH]
c:0032 p:---- s:0150 e:000149 CFUNC  :map!
c:0031 p:0007 s:0146 E:000728 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:169
c:0030 p:0005 s:0141 E:0002a0 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:147
c:0029 p:0014 s:0135 E:0008d0 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:204
c:0028 p:0004 s:0129 E:001ed0 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:51 [FINISH]
c:0027 p:---- s:0126 e:000125 CFUNC  :class_eval
c:0026 p:0017 s:0122 E:000d48 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/helpers.rb:51
c:0025 p:0006 s:0117 E:000918 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionview-6.1.1/lib/action_view/layouts.rb:219
c:0024 p:0007 s:0112 E:001eb0 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/actionpack-6.1.1/lib/abstract_controller/railties/routes_helpers.rb:9 [FINISH]
c:0023 p:0019 s:0107 E:000760 TOP    /Users/shep/Projects/repro/app/mailers/application_mailer.rb:1 [FINISH]
c:0022 p:---- s:0104 e:000103 CFUNC  :require
c:0021 p:0035 s:0099 E:001990 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26 [FINISH]
c:0020 p:---- s:0093 e:000092 CFUNC  :const_get
c:0019 p:0056 s:0087 E:000b80 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409
c:0018 p:0047 s:0080 E:000a60 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:733 [FINISH]
c:0017 p:---- s:0075 e:000074 CFUNC  :foreach
c:0016 p:0014 s:0070 E:001c68 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725
c:0015 p:0043 s:0065 E:001650 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:404 [FINISH]
c:0014 p:---- s:0058 e:000057 CFUNC  :synchronize
c:0013 p:0006 s:0054 E:000340 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:393
c:0012 p:0010 s:0050 E:001410 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/railties-6.1.1/lib/rails/autoloaders.rb:30
c:0011 p:0031 s:0046 E:002570 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/railties-6.1.1/lib/rails/application.rb:510
c:0010 p:0041 s:0042 E:000ae8 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/sorbet-0.5.6281/lib/require_everything.rb:28
c:0009 p:0019 s:0037 E:001ac0 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/sorbet-0.5.6281/lib/require_everything.rb:18
c:0008 p:0020 s:0033 E:000a30 BLOCK  /Users/shep/.rvm/gems/ruby-3.0.0/gems/sorbet-0.5.6281/lib/gem-generator-tracepoint.rb:39
c:0007 p:0006 s:0030 E:001b40 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/sorbet-0.5.6281/lib/gem-generator-tracepoint/tracer.rb:80
c:0006 p:0023 s:0026 E:0020b0 METHOD /Users/shep/.rvm/gems/ruby-3.0.0/gems/sorbet-0.5.6281/lib/gem-generator-tracepoint.rb:38
```

I do not see any relevant logs in `DiagnosticReports`, but that may be a system configuration setting of mine.

#### Expected behavior

It would not segfault.

- - -

```
% ruby --version
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin20]
```

```
% rails --version
Rails 6.1.1
```

```
% gem list | grep sorbet
sorbet (0.5.6281)
sorbet-coerce (0.3.0)
sorbet-rails (0.7.3)
sorbet-runtime (0.5.6281)
sorbet-static (0.5.6281 universal-darwin-20)
```

macOS 11.2 (20D64) (x86_64)

**Comment by shepmaster** [comment]
I added [sorbet-rails](https://github.com/chanzuckerberg/sorbet-rails) and reran. Still a segfault, but this one finished producing output (big, so I can't inline them):

- [output.zip](https://github.com/sorbet/sorbet/files/5954106/output.zip)
- [crash.log.zip](https://github.com/sorbet/sorbet/files/5954105/crash.log.zip)

**Comment by shepmaster** [comment]
Ah, I see now that Sorbet doesn't yet support Ruby 3 — https://github.com/sorbet/sorbet/issues/3851. I guess the various blog posts I read confused me!

**Comment by jez** [comment]
Hey! Thanks for the report. sorbet-runtime does not have any C extension code that could cause a segmentation fault.

If you have managed to get the Ruby VM to segmentation fault, it is a bug in upstream Ruby or some other gem. The Ruby core team treats it as a guarantee that "normal" Ruby (i.e., code implemented with Ruby source code, not C extensions) will not segfault.

Nice job finding a bug in Ruby!

**Comment by peterzhu2118** [comment]
Thank you for the bug report. It should be fixed in ruby/ruby#4175 which has been flagged for backporting so it should be included in Ruby 3.0.1.
